torch/_inductor/fx_passes/post_grad.py

```diff
@@ -354,6 +354,38 @@ def shape_of_mm(a, b):
     return [m, n]
 
 
+# TODO: this pattern is generalizable to all accumulate funcions
```

Comment:
Even more, this pattern is generalisable to any linear function! :)

Reply:
Given the generality, it may be cool to implement the general pattern, but I think that it'll mostly be useful for indexing ops I think. @peterbell10 thoughts?

Reply:
First thought, this pattern isn't numerically equivalent (which has been on the brain recently).

Second I wonder does this pattern or ones like it come up a lot? I could imagine for a tensor that's indexed multiple times this might appear in the backward graph. Would be good to see some context of how it fits in.

Reply:
Yes, it isn't equivalent in floating point arithmetic. Not sure what to do here.

For context, this came up in https://github.com/pytorch/pytorch/pull/121769

Reply:
I wouldn't worry too much about the asociativity here, as `index_put` itself is non-deterministic for the same reason.